hepcrawl/parsers/jats.py

```diff
@@ -455,3 +465,65 @@ def get_author(self, author_node):
         affiliations = self.get_author_affiliations(author_node)
 
         return self.builder.make_author(author_name, raw_affiliations=affiliations, emails=emails)
+
+    @staticmethod
+    def get_reference_authors(ref_node, role):
+        """Extract authors of `role` from a reference node.
+
+        Args:
+            ref_node(scrapy.selector.Selector): a selector on a single reference.
+            role(str): author role
+
+        Returns:
+            List[str]: list of names
+        """
+        return ref_node.xpath(
+            './person-group[@person-group-type=$role]/string-name/text()',
+            role=role
+        ).extract()
+
+
+    def get_reference(self, ref_node):
+        """Extract one reference.
+
+        Args:
+            citation_node(scrapy.selector.Selector): a selector on a single
+                reference, i.e. ``<ref>``.
+
+        Returns:
+            dict: the parsed reference, as generated by
+                :class:`inspire_schemas.api.ReferenceBuilder`
+        """
+
+        builder = ReferenceBuilder()
+
+        citation_node = ref_node.xpath('./mixed-citation')
+
+        fields = [
+            ('./source/text()', builder.set_journal_title),
+            ('./volume/text()', builder.set_journal_volume),
+            ('./issue/text()', builder.set_journal_issue),
+            ('./year/text()', builder.set_year),
+            ('./pub-id[@pub-id-type="arxiv"]/text()', builder.add_uid),
+            ('./pub-id[@pub-id-type="doi"]/text()', builder.add_uid),
+            ('./article-title/text()', builder.add_title),
+            ('../label/text()', lambda x: builder.set_label(x.strip('[].')))
+        ]
+
+        for xpath, handler in fields:
+            value = citation_node.xpath(xpath).extract_first()
+            if value:
+                handler(value)
+
+        for author in self.get_reference_authors(citation_node, 'author'):
+            builder.add_author(author, 'author')
+
+        for editor in self.get_reference_authors(citation_node, 'editor'):
```

Comment:
I would swap this with the authors, to have the editors before the authors in the record.